components/dashboard/src/settings/ProfileInformation.tsx

```diff
@@ -0,0 +1,208 @@
+            before.company !== after.company ||
+            before.avatarURL !== after.avatarURL
+        );
+    }
+
+    function shouldNudgeForUpdate(user: User): boolean {
+        if (!user.additionalData) {
+            // never updated profile information and account is older than 24 hours (i.e. ask on second day).
+            return !isDateSmaller(hoursBefore(new Date().toISOString(), 24), user.creationDate);
+        }
+        // if the profile wasn't updated for 12 months ask again.
+        return !(
+            !!user.additionalData.lastUpdatedDetailsNudge &&
+            isDateSmaller(hoursBefore(new Date().toISOString(), 24 * 365), user.additionalData.lastUpdatedDetailsNudge)
+        );
+    }
+
+    /**
+     * @param state
+     * @returns error message or empty string when valid
+     */
+    export function validate(state: ProfileState): string {
+        if (state.name.trim() === "") {
+            return "Name must not be empty.";
+        }
+        if (state.email.trim() === "") {
+            return "Email must not be empty.";
+        }
+        if (
+            !/^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(
+                state.email.trim(),
+            )
+        ) {
+            return "Please enter a valid email.";
+        }
+        return "";
+    }
+
+    export function NudgeForProfileUpdateModal() {
+        const { user, setUser } = useContext(UserContext);
+        const original = ProfileState.getProfileState(user!);
+        const [profileState, setProfileState] = useState(original);
+        const [errorMessage, setErrorMessage] = useState("");
+        const [visible, setVisible] = useState(shouldNudgeForUpdate(user!));
+
+        const saveProfileState = () => {
+            const error = ProfileState.validate(profileState);
+            setErrorMessage(error);
+            if (error) {
+                return;
+            }
+            const updatedUser = ProfileState.setProfileState(user!, profileState);
+            setUser(updatedUser);
+            getGitpodService().server.updateLoggedInUser(updatedUser);
+            setVisible(shouldNudgeForUpdate(updatedUser!));
+        };
+
+        const cancelProfileUpdate = () => {
+            setProfileState(original);
+            saveProfileState();
+        };
+
+        return (
+            <Modal
+                title="Update Profile Information"
+                visible={visible}
+                onClose={cancelProfileUpdate}
+                closeable={true}
+                className="_max-w-xl"
+                buttons={
+                    <div>
+                        <button className="secondary" onClick={cancelProfileUpdate}>
+                            Dismiss
+                        </button>
+                        <button className="ml-4" onClick={saveProfileState}>
```

Comment:
**nitpick**: Spacing fix according to https://github.com/gitpod-io/gitpod/pull/11023#discussion_r911144123, following the spacing pattern used across the product to separate related elements. `ml-4` can be used to separate unrelated elements..

```suggestion
                        <button className="ml-2" onClick={saveProfileState}>
```